XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandControllerDataProvider.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.Providers.Controllers;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public class HandControllerDataProvider : BaseControllerDataProvider, IMixedRealityHandControllerDataProvider
+    {
+        private IMixedRealityHand leftHand;
+        private IMixedRealityHand rightHand;
```

Comment:
Should these be implemented in `IMixedRealityHandControllerDataProvider`?

Reply:
Hmm for now I'd say no. These two fields are recalculated every frame in LateUpdate and for the data provider's internal use.